src/core/font_substitutions.js

```diff
@@ -333,38 +287,96 @@ const substitutionMap = new Map([
 
 const fontAliases = new Map([["Arial-Black", "ArialBlack"]]);
 
+function getStyleToAppend(style) {
+  switch (style) {
+    case BOLD:
+      return "Bold";
+    case ITALIC:
+      return "Italic";
+    case BOLDITALIC:
+      return "Bold Italic";
+    default:
+      if (!style) {
+        return "";
+      }
+      if (style.weight === "bold") {
+        return "Bold";
+      }
+      if (style.style === "italic") {
+        return "Italic";
+      }
```

Comment:
Given how this is a fallback code-path, we could perhaps shorten the code a little bit?
```suggestion
      if (style?.weight === "bold") {
        return "Bold";
      }
      if (style?.style === "italic") {
        return "Italic";
      }
```